integration-tests/src/test/java/com/ozonehis/eip/odoo/openmrs/MedicationRequestToSaleOrderIntegrationTest.java

```diff
@@ -0,0 +1,175 @@
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.openmrs.eip.fhir.Constants.HEADER_FHIR_EVENT_TYPE;
+
+import com.ozonehis.eip.odoo.openmrs.client.OdooUtils;
+import com.ozonehis.eip.odoo.openmrs.model.Partner;
+import com.ozonehis.eip.odoo.openmrs.model.SaleOrder;
+import com.ozonehis.eip.odoo.openmrs.model.SaleOrderLine;
+import com.ozonehis.eip.odoo.openmrs.routes.partner.CreatePartnerRoute;
+import com.ozonehis.eip.odoo.openmrs.routes.partner.DeletePartnerRoute;
+import com.ozonehis.eip.odoo.openmrs.routes.partner.UpdatePartnerRoute;
+import com.ozonehis.eip.odoo.openmrs.routes.saleorder.CreateSaleOrderRoute;
+import com.ozonehis.eip.odoo.openmrs.routes.saleorder.DeleteSaleOrderRoute;
+import com.ozonehis.eip.odoo.openmrs.routes.saleorder.UpdateSaleOrderRoute;
+import com.ozonehis.eip.odoo.openmrs.routes.saleorderline.CreateSaleOrderLineRoute;
+import com.ozonehis.eip.odoo.openmrs.routes.saleorderline.DeleteSaleOrderLineRoute;
+import com.ozonehis.eip.odoo.openmrs.routes.saleorderline.UpdateSaleOrderLineRoute;
+import java.net.MalformedURLException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.camel.CamelContext;
+import org.apache.camel.test.infra.core.annotations.RouteFixture;
+import org.apache.xmlrpc.XmlRpcException;
+import org.hl7.fhir.r4.model.Bundle;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+public class MedicationRequestToSaleOrderIntegrationTest extends BaseRouteIntegrationTest {
+
+    private Bundle medicationRequestBundle;
+
+    private static final String ENCOUNTER_PART_OF_UUID = "26616e46-2cfe-4563-afaa-c243ca94f4c7";
+
+    private static final String PATIENT_UUID = "79355a93-3a4f-4490-98aa-278f922fa87c";
+
+    @BeforeEach
+    public void initializeData() {
+        medicationRequestBundle = loadResource("fhir.bundle/medication-request-bundle.json", new Bundle());
+    }
+
+    @RouteFixture
+    public void createRouteBuilder(CamelContext context) throws Exception {
+        context = getContextWithRouting(context);
+
+        context.addRoutes(new CreatePartnerRoute());
+        context.addRoutes(new UpdatePartnerRoute());
+        context.addRoutes(new DeletePartnerRoute());
+        context.addRoutes(new CreateSaleOrderRoute());
+        context.addRoutes(new UpdateSaleOrderRoute());
+        context.addRoutes(new DeleteSaleOrderRoute());
+        context.addRoutes(new CreateSaleOrderLineRoute());
+        context.addRoutes(new UpdateSaleOrderLineRoute());
+        context.addRoutes(new DeleteSaleOrderLineRoute());
+    }
+
+    @Test
+    @DisplayName("Should verify has sale routes.")
+    public void shouldVerifySaleOrderRoutes() {
+        assertTrue(hasRoute(contextExtension.getContext(), "medication-request-to-sale-order-router"));
```

Comment:
Shouldn't you be calling first something before verifying anything? Otherwise the test if not intuitive

Reply:
I believe the `@RouteFixture` takes care of that. Here is just verifying if the routes are available(registered) in the camelContext.